core/src/main/java/io/questdb/cutlass/http/processors/JsonQueryProcessorState.java

```diff
@@ -308,16 +308,36 @@ private static void putTimestampValue(HttpChunkedResponseSocket socket, Record r
         socket.put('"').putISODate(t).put('"');
     }
 
-    private static void putGeoHashStringValue(HttpChunkedResponseSocket socket, Record rec, int col, int precision) {
-        final long l = rec.getGeoHash(col);
-        if (l == GeoHashes.NULL) {
+    private static void putGeoHashStringByteValue(HttpChunkedResponseSocket socket, Record rec, int col, int columnType) {
+        byte l = rec.getGeoHashByte(col);
+        putGeoHashStringValue(socket, columnType, l);
+    }
+
+    private static void putGeoHashStringShortValue(HttpChunkedResponseSocket socket, Record rec, int col, int columnType) {
+        short l = rec.getGeoHashShort(col);
+        putGeoHashStringValue(socket, columnType, l);
+    }
+
+    private static void putGeoHashStringIntValue(HttpChunkedResponseSocket socket, Record rec, int col, int columnType) {
+        int l = rec.getGeoHashInt(col);
+        putGeoHashStringValue(socket, columnType, l);
+    }
+
+    private static void putGeoHashStringLongValue(HttpChunkedResponseSocket socket, Record rec, int col, int columnType) {
+        long l = rec.getGeoHashLong(col);
+        putGeoHashStringValue(socket, columnType, l);
+    }
+
+    private static void putGeoHashStringValue(HttpChunkedResponseSocket socket, int columnType, long value) {
+        final int precision = GeoHashes.getBitsPrecision(columnType);
+        if (value == GeoHashes.NULL) {
             socket.put("null");
         } else {
             socket.put('\"');
-            if (precision < 0) {
-                GeoHashes.toString(l, -precision, socket);
+            if (precision % 5 == 0) {
```

Comment:
could we undo this optimisation regression pls ?

Reply:
No, I don't see a clean way.
Clean type value is needed to call correct geohash retrieval method. Pre-computing binary vs string flag and fetching it from another array is the option but will it really be faster than %5, /5 ?

Reply:
I changed it so that it "may" be as "quick" as it was by introducing another int per column, it's the best I can think of